src/garage/torch/algos/td3.py

```diff
@@ -0,0 +1,429 @@
+        self._replay_buffers = PathBuffer(
+            capacity_in_transitions=self._replay_buffer_size)
+
+    def _get_action(self, action, noise_scale):
+        """Select action based on policy.
+
+        Action can be added with noise.
+
+        Args:
+            action (float): Action.
+            noise_scale (float): Noise scale added to action.
+
+        Return:
+            float: Action selected by the policy.
+        """
+        action += noise_scale * np.random.randn(self._action_dim)
+        return np.clip(action, -self._max_action, self._max_action)
+
+    def train(self, trainer):
+        """Obtain samplers and start actual training for each epoch.
+
+        Args:
+            trainer (Trainer): Experiment trainer, which provides services
+                such as snapshotting and sampler control.
+
+        Returns:
+            float: The average return in last epoch cycle.
+        """
+        if not self._eval_env:
+            self._eval_env = trainer.get_env_copy()
+        last_returns = None
+        trainer.enable_logging = False
+        for _ in trainer.step_epochs():
+            for cycle in range(self._steps_per_epoch):
+                # Obtain trasnsition batch and store it in replay buffer.
+                # Get action randomly from environment within warm-up steps.
+                # Afterwards, get action from policy.
+                if self._uniform_random_policy and \
+                        trainer.step_itr < self._start_steps:
+                    trainer.step_path = trainer.obtain_episodes(
+                        trainer.step_itr,
+                        agent_update=self._uniform_random_policy)
+                else:
+                    trainer.step_path = trainer.obtain_episodes(
+                        trainer.step_itr, agent_update=self.exploration_policy)
+                self._replay_buffer.add_episode_batch(trainer.step_path)
+
+                # Update after warm-up steps.
+                if trainer.total_env_steps >= self._update_after:
+                    self._train_once(trainer.step_itr)
+
+                # Evaluate and log the results.
+                if (cycle == 0 and self._replay_buffer.n_transitions_stored >=
+                        self._min_buffer_size):
+                    trainer.enable_logging = True
+                    eval_eps = self._evaluate_policy()
+                    log_performance(trainer.step_path,
+                                    eval_eps,
+                                    discount=self._discount,
+                                    prefix='Training')
+                    last_returns = log_performance(trainer.step_itr,
+                                                   eval_eps,
+                                                   discount=self._discount,
+                                                   prefix='Evaluation')
+                trainer.step_itr += 1
+
+        return np.mean(last_returns) if last_returns else 0
```

Comment:
I still don't think we should return anything from `.train()`, since it's not statistically meaningful, and tests using it tend to be flaky.